treeherder/client/thclient/client.py

```diff
@@ -684,6 +686,14 @@ def _get_uri(self, project, endpoint, data=None, oauth_key=None,
 
         return uri
 
+    def _get_revision_id(self, repo_name, revision):
+        """
+        Return the id of a given revision from treeherder
+        """
+        url = "%s/%s/resultset/?revision=%s" % (self.API_ROOT, repo_name, revision)
```

Comment:
Actually I already wrote a _get_json method which you can copy (I might add it as part of bug 1163138):

    def _get_json(self, project, endpoint):
        resp = requests.get(self._get_uri(project, endpoint), timeout=self.timeout)
        resp.raise_for_status()
        return resp.json()